Minor changes from febr.

benhrisdev opened the pull request.

benhrisdev committed:
adding new module which helps integrate odoo's hr tools with Neokgindom's external website

benhrisdev committed:
adding new module where general changes are being made to odoo what can be applicabe amongst most of the neocommonwealth projects

PeterSchwarzHPI commented on addons_neokingdom/neok_common_project_mods/views/project_project.xml:
```diff
@@ -0,0 +1,30 @@
+    </record>
+
+    <!-- Hiding some not needed field from list to prevent confusion -->
+    <record id="neok_proj_view_project_mod" model="ir.ui.view">
```
Let's keep the initial id ... it's uniquely identifiable by the module prefix.

benhrisdev replied:
its for maintenance purposes, this name is more easily searchable.

PeterSchwarzHPI commented on addons_neokingdom/neok_common_project_mods/views/project_project.xml:
```diff
@@ -0,0 +1,30 @@
+
+    <!-- Hiding some not needed field from list to prevent confusion -->
+    <record id="neok_proj_view_project_mod" model="ir.ui.view">
+        <field name="name">neok_proj_view_project_mod</field>
```
Same for name (maybe suffix _neok)

PeterSchwarzHPI commented on addons_neokingdom/neok_hr_expose_for_website/controllers/main.py:
```diff
@@ -0,0 +1,147 @@
+
+            job_id = request.env['hr.job'].sudo().browse(int(job_id))
+
+            if not job_id.exists():
+                return Response(
+                    json.dumps({'msg': 'Position not found.'}),
+                    status=404,
+                    mimetype='application/json'
+                )
+            if not job_id.website_published:
+                return Response(
+                    json.dumps({'msg': 'Position is closed!'}),
+                    status=400,
+                    mimetype='application/json'
+                )
```
Use a method for "message, code" to return the response ... and only use one return statement for the exception case.

_check_error_message e.g.

benhrisdev replied:
it saves 2 lines but doesnt reduce to much repetition, since its just creating an object. but im fine with it.

PeterSchwarzHPI reviewed (commented):
Let's not prefix modules with neok.
Let's only do it if we adapt standard modules (e.g. project views -> neok_project).

But if you create a new module with new functionality ... please just give it a general naming

benhrisdev committed:
details:
- changes based on pull request: https://github.com/NeokingdomDAO/odoo/pull/20
- addon folders restructure
- modified "My Projects" filter based on contributions
- minor module content restructure (naming, manifest, etc..)

benhrisdev committed:
minor hr website expose module refactor
